test/sandbox/sandbox.go

```diff
@@ -641,34 +658,82 @@ func buildTree(t testing.TB, root *config.Root, layout []string) {
 	}
 
 	for _, spec := range layout {
-		path, data := parsePathData(spec)
-
-		specKind := string(spec[0:2])
+		param1, param2 := parseParams(spec)
+		colonIndex := strings.Index(spec, ":") + 1
+		specKind := string(spec[0:colonIndex])
 		switch specKind {
-		case "d:":
-			test.MkdirAll(t, filepath.Join(rootdir, spec[2:]))
-		case "l:":
-			target := filepath.Join(rootdir, path)
-			linkName := filepath.Join(rootdir, data)
+		case "d:", "dir:":
+			test.MkdirAll(t, filepath.Join(rootdir, spec[colonIndex:]))
+		case "l:", "link:":
+			target := filepath.Join(rootdir, param1)
+			linkName := filepath.Join(rootdir, param2)
 			test.Symlink(t, target, linkName)
-		case "g:":
-			repodir := filepath.Join(rootdir, spec[2:])
+		case "g:", "git:":
+			repodir := filepath.Join(rootdir, spec[colonIndex:])
 			test.MkdirAll(t, repodir)
 			git := NewGit(t, repodir)
 			git.Init()
-		case "s:":
-			if data == "" {
-				abspath := filepath.Join(rootdir, path)
+		case "s:", "stack:":
+			if param2 == "" {
+				abspath := filepath.Join(rootdir, param1)
 				stackdir := project.PrjAbsPath(rootdir, abspath)
 				assert.NoError(t, stack.Create(root, config.Stack{Dir: stackdir}))
 				continue
 			}
 
-			gentmfile(path, data)
-		case "f:":
-			test.WriteFile(t, rootdir, path, data)
+			genStackFile(param1, param2)
+		case "f:", "file:":
+			test.WriteFile(t, rootdir, param1, param2)
+		case "copy:":
+			assert.NoError(t, fs.CopyAll(
+				filepath.Join(rootdir, param1),
+				param2,
+			))
+		case "run:":
+			cmd := exec.Command("sh", "-c", param2)
```

Comment:
Do we assume sh is always present, even on Windows? If not, could make this more portable in the future.

Reply:
nice catch! no we should not assume that! I forgot to improve this. Thanks